File: AoC2019/JavaScript/day2.js
//puzzle setup
const fs = require('fs');
inputs = fs.readFileSync('../day2in.txt','utf8');

//do stuff
arr = inputs.split(',').map(Number);
i = 0;
op = arr[0];

while (op != 99)
{
    op = arr[i];
    if (op == 1) arr[arr[i+3]] = arr[arr[i+1]] + arr[arr[i+2]];
    if (op == 2) arr[arr[i+3]] = arr[arr[i+1]] * arr[arr[i+2]];
    i += 4;
}
console.log(arr[0]);

File: AoC2019/Python3/day2.py
#puzzle setup
with open('./day2in.txt','r') as pInput:
    pInput = pInput.read().split(',')
pInput = [int(x) for x in pInput]

#per puzzle instructions..
pInput[1] = 12
pInput[2] = 2

#do stuff
for i in range(0,len(pInput),4):
    opcode = pInput[i]
    valA = pInput[i+1]
    valB = pInput[i+2]
    resultPos = pInput[i+3]

    if (opcode == 99):
        print('Exit opcode 99, position 0 output: ' + str(pInput[0]))
        quit()
    elif (opcode == 1):
        pInput[resultPos] = pInput[valA] + pInput[valB]
    elif (opcode == 2):
        pInput[resultPos] = pInput[valA] * pInput[valB]
    else:
        print('Error: unknown opcode ' + str(opcode) + ' at position ' + str(i))
        quit()
